"panic: runtime error" when deploying a service with 0 replicas

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/docker/compose-cli/blob/master/CONTRIBUTING.md#reporting-other-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- Docker Support Forums - https://forums.docker.com
- Docker Community Slack - https://dockr.ly/community
- Post a question on StackOverflow, using the Docker tag

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**
"panic: runtime error" when deploying a service with 0 replicas

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

**Steps to reproduce the issue:**
1. Run `docker compose up` with the following docker-compose.yml
```yaml
services:
  nginx1:
    image: nginx:alpine
    deploy:
      replicas: 0
  nginx2:
    image: nginx:alpine
```

**Describe the results you received:**
```
❯ docker compose up
[+] Running 0/1
 ⠹ Network docker-compose_default  Creating                                                        0.3s
panic: runtime error: index out of range [0] with length 0

goroutine 56 [running]:
github.com/docker/compose-cli/pkg/compose.(*convergence).updateProject(0xc00000cfd8, 0xc000469b80, 0xc0001bdfa0, 0x5)
        github.com/docker/compose-cli/pkg/compose/convergence.go:114 +0xfcf
github.com/docker/compose-cli/pkg/compose.(*convergence).apply.func1(0x20d6548, 0xc00037d260, 0xc0001bdfa0, 0x5, 0x1f2d4e5, 0x8)
        github.com/docker/compose-cli/pkg/compose/convergence.go:104 +0x292
github.com/docker/compose-cli/pkg/compose.run.func1(0x0, 0x0)
        github.com/docker/compose-cli/pkg/compose/dependencies.go:102 +0xa3
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc00037de30, 0xc0005ad560)
        golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
        golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x66
```

**Describe the results you expected:**
Service nginx2 gets started, service nginx1 does not.

**Additional information you deem important (e.g. issue happens only occasionally):**
Introduced after upgrading to Docker for Mac 3.5.2 (Docker compose version v2.0.0-beta.6), it works correctly with Docker for Mac 3.5.1 (Docker compose version v2.0.0-beta.4).

**Output of `docker-compose --version`: ***
```
Docker Compose version v2.0.0-beta.6
```

**Output of `docker version`:**

```
Client:
 Cloud integration: 1.0.17
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.16.4
 Git commit:        f0df350
 Built:             Wed Jun  2 11:56:22 2021
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       b0f5bc3
  Built:            Wed Jun  2 11:54:58 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.6
  GitCommit:        d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc:
  Version:          1.0.0-rc95
  GitCommit:        b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker context show`:**  
You can also run `docker context inspect context-name` to give us more details but don't forget to remove sensitive content.

```
default
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  compose: Docker Compose (Docker Inc., v2.0.0-beta.6)
  scan: Docker Scan (Docker Inc., v0.8.0)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 66
 Server Version: 20.10.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2 io.containerd.runtime.v1.linux
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc version: b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.25-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 7.774GiB
 Name: docker-desktop
 ID: FMLZ:6CFC:Q5KC:JHI3:JQJK:2T7Z:G52D:HYBV:X42U:425C:RYZI:E6IS
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details (AWS ECS, Azure ACI, local, etc.):**
local

--- a/pkg/compose/convergence.go
+++ b/pkg/compose/convergence.go
@@ -111,6 +111,9 @@ var mu sync.Mutex
 // updateProject updates project after service converged, so dependent services relying on `service:xx` can refer to actual containers.
 func (c *convergence) updateProject(project *types.Project, service string) {
 	containers := c.getObservedState(service)
+	if len(containers) == 0 {
+		return
+	}
 	container := containers[0]
 
 	// operation is protected by a Mutex so that we can safely update project.Services while running concurrent convergence on services
@@ -557,6 +560,9 @@ func (s *composeService) startService(ctx context.Context, project *types.Projec
 	}
 
 	if len(containers) == 0 {
+		if scale, err := getScale(service); err != nil && scale == 0 {
+			return nil
+		}
 		return fmt.Errorf("no containers to start")
 	}
 

